frontend/js/marc-record-merge-postmerge-service.js

```diff
@@ -478,14 +478,29 @@ export function add080VersionCode (preferredRecord, otherRecord, mergedRecordPar
         }
       });
       if (!hasCode2 && hasCode9FKeep) {
-        field.subfields.push({ code: '2', value: '1974/fin/fennica' });
-        field.subfields.sort(sortSubfields);
+        field.subfields = pushSubfieldInOrder(field);
       }
     });
 
   return { 
     mergedRecord
   };
+
+  function pushSubfieldInOrder(field) {
```

Comment:
Refactor as follows:
1. Check if the subfield already exists (Code === `2` and value === `1974/fin/fennica`)
2. If it doesn't, find the correct index to insert the subfield to (You can use [Array.prototype.findIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex))
3. Insert the subfield (You can use[ Array.prototype.splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice))